### ISSUE TITLE
Unable to retrieve indexes

### ISSUE THREAD

**Issue by carldaws** [ankane/neighbor-redis#1]
If an index is created but the reference to it is lost (either by reassignment, or between sessions), it is not possible to retrieve it for further appending to or querying. This makes it very difficult (impossible?) to manage multiple indexes, for example where each instance of a Rails model may have its own index to query.

I have started on a very rudimentary implementation of a RetrievedIndex class (see [here](https://github.com/cdwsn/neighbor-redis/tree/retrieve-indexes)) but I'm struggling to find a way to recapture the parameters passed to Redis at index creation time (for instance FT.INFO doesn't seem to return much of use to the Index class) and so a general solution is evading me right now.

Any help is appreciated!

**Comment by ankane** [comment]
Hi @cdwsn, the way to retrieve an index is to initialize an index object.

```ruby
index = Neighbor::Redis::HNSWIndex.new("items", dimensions: 3, distance: "l2")
```

The current design expects index parameters to be kept in code (rather than stored in Redis on index creation, which would require an additional query to Redis on instantiation).

**Comment by carldaws** [comment]
Hi @ankane 

I've abandoned the previous approach and added serialization / deserialization methods (and a couple of tests) to [this branch](https://github.com/cdwsn/neighbor-redis/tree/serialized-indexes) of my fork. This is a bit of a quick fix, but it'll do just fine for my needs. 

Another approach is to allow the definition of index parameters ahead of time in a config, and then use those to create and retrieve indexes by name alone. I've implemented that on [this branch](https://github.com/cdwsn/neighbor-redis/tree/config). 

I haven't yet decided which approach is best for my specific use case, but let me know if you'd be open to a pull request for either to your repo.
